src/include/wiredtiger.in

```diff
@@ -446,6 +478,38 @@ struct __wt_cursor {
 	int __F(insert)(WT_CURSOR *cursor);
 
 	/*!
+	 * Modify an existing record.
+	 *
+	 * Both the key and value must be set and the record must already exist;
+	 * the record will be updated.
+	 *
+	 * Modification structures are applied in order, and later modifications
+	 * can update earlier modifications.
+	 *
+	 * The modify method is only supported on raw byte arrays accessed using
+	 * a WT_ITEM structure, that is, a format type of \c u.
+	 *
+	 * @snippet ex_all.c Modify an existing record
+	 *
+	 * On success, the cursor ends positioned at the modified record; to
+	 * minimize cursor resources, the WT_CURSOR::reset method should be
+	 * called as soon as the cursor no longer needs that position.
+	 *
+	 * The maximum length of a single column stored in a table is not fixed
+	 * (as it partially depends on the underlying file configuration), but
+	 * is always a small number of bytes less than 4GB.
+	 *
+	 * @param cursor the cursor handle
+	 * @param entries an array of modification data structures
+	 * @param nentries the number of modification data structures
+	 * @errors
+	 * In particular, if \c in_memory is configured for the database and
+	 * the modify requires more than the configured cache size to complete,
+	 * ::WT_CACHE_FULL is returned.
+	 */
+	int __F(modify)(WT_CURSOR *cursor, WT_MODIFY *entries, int nentries);
```

Comment:
This is roughly what I originally proposed (although I wanted a new set-value API, 
 called `WT_CURSOR.set_value_partial`, that took an array of `WT_MODIFY` structures).

I still think it's more natural in WiredTiger to have `WT_CURSOR.set_value_partial` plus `WT_CURSOR.update`, rather than bundle setting and updating the value into `WT_CURSOR.modify`.

If a new value can be passed to `WT_CURSOR.modify`, why can't a new value be passed to in `WT_CURSOR.update`?